www/src/pages/headless/combobox.md

```diff
@@ -0,0 +1,432 @@
+
+    // set up (two-way) data binding
+    value(country)
+
+    comboboxInput {}
+
+    comboboxItems("bg-white") {
+        // using a combination of 'render' and 'forEach' is the intended
+        // way of populating the dropdown
+        results.render { result ->
+            result.items.forEach { item ->
+                comboboxItem(item) {
+                    span {
+                        className(active.map { if (it) "underline" else "" })
+                        +item.value
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+::: info
+Do not forget to include a `portalRoot()`-call at the end of your initial `RenderContext` as explained
+[here](/headless/#portalling)!
+:::
+
+## Styling the active or selected Item
+
+A `comboboxItem` has two special states:
+
+- When the mouse is hovering above it or the user navigates to the item via keyboard, it is considered `active`.
+- A click or [[Enter]] selects the active item. It is then `selected`.
+
+In the scope of a `comboboxItem`, `active` and `selected` are available as a `Flow<Boolean>` to apply styling or to
+render or hide certain elements based on the state:
+
+```kotlin
+comboboxItem(item) {
+    //change fore- und background-color is item is active 
+    className(active.map {
+        if (it) "text-amber-900 bg-amber-100" else "text-gray-300"
+    })
+
+    span { +item.value }
+
+    //render checked-icon only if item is selected
+    selected.render {
+        if (it) {
+            svg { content(HeroIcons.check) }
+        }
+    }
+}
+```
+
+## Highlight matching items
+
+When rendering items based on the user's query, it is often a requirement to highlight the passages of each
+item's String representation that match the given query.
+For this purpose, the current `query` is part of the `results` Flow.
+
+```kotlin
+comboboxItems {
+    results.render { (query, items, _) ->
+        items.forEach { item ->
+            comboboxItem(item) {
+                // separate the item's String representation into matching and
+                // non-matching segments
+                val segments = item.value.split(
+                    Regex(
+                        "(?<=($highlight))|(?=($highlight))",
+                        RegexOption.IGNORE_CASE
+                    )
+                )
+
+                for (s in segments) {
+                    span(
+                        joinClasses(
+                            "underline".takeIf {
+                                s.contentEquals(query, ignoreCase = true)
+                            }
+                        )
+                    ) { +s }
+                }
+            }
+        }
+    }
+}
+```
+
+## Truncated result list
+
+The `results` Flow only displays a fixed number of items at a time, configured via the `maximumDisplayedItems` property.
+If more items match the given query than configured to be displayed, the `truncated` property available in the `results`
```

Comment:
perhaps name it flag instead of property here? "... the truncated flag available ..."